packages/teleport/src/Discover/Shared/SetupAccess/SetupAccessWrapper.tsx

```diff
@@ -0,0 +1,154 @@
+ * limitations under the License.
+ */
+
+import React from 'react';
+import styled from 'styled-components';
+import { Text, Box, Indicator } from 'design';
+import * as Icons from 'design/Icon';
+
+import {
+  Header,
+  HeaderSubtitle,
+  ActionButtons,
+  ButtonBlueText,
+} from 'teleport/Discover/Shared';
+
+import { AccessInfo } from './AccessInfo';
+
+import type { TraitKind } from './AccessInfo';
+import type { State } from './useUserTraits';
+
+export type Props = {
+  isSsoUser: State['isSsoUser'];
+  canEditUser: State['canEditUser'];
+  attempt: State['attempt'];
+  fetchUserTraits: State['fetchUserTraits'];
+  headerSubtitle: string;
+  traitKind: TraitKind;
+  traitDescription: string;
+  hasTraits: boolean;
+  onProceed(): void;
+  children: React.ReactNode;
+  infoContent?: React.ReactNode;
+};
+
+export function SetupAccessWrapper({
+  attempt,
+  fetchUserTraits,
+  canEditUser,
+  isSsoUser,
+  hasTraits,
+  traitKind,
+  traitDescription,
+  headerSubtitle,
+  onProceed,
+  children,
+  infoContent,
+}: Props) {
+  const canAddTraits = !isSsoUser && canEditUser;
+
+  let $content;
+  switch (attempt.status) {
+    case 'failed':
+      $content = (
+        <>
+          <Text my={3}>
+            <Icons.Warning ml={1} mr={2} color="danger" />
+            Encountered Error: {attempt.statusText}
+          </Text>
+          <ButtonBlueText ml={1} onClick={fetchUserTraits}>
+            Retry
+          </ButtonBlueText>
+        </>
+      );
+      break;
+
+    case 'processing':
+      $content = (
+        <Box mt={4} textAlign="center" height="70px" width="300px">
+          <Indicator delay="none" />
+        </Box>
+      );
+      break;
+
+    case 'success':
+      if (isSsoUser && !hasTraits) {
+        $content = (
+          <AccessInfo
+            accessKind="ssoUserAndNoTraits"
+            traitKind={traitKind}
+            traitDesc={traitDescription}
+          />
+        );
+      } else if (!canAddTraits && !hasTraits) {
+        $content = (
+          <AccessInfo
+            accessKind="noAccessAndNoTraits"
+            traitKind={traitKind}
+            traitDesc={traitDescription}
+          />
+        );
+      } else {
+        $content = (
+          <>
+            <StyledBox>{children}</StyledBox>
+            {!isSsoUser && !canEditUser && (
+              <AccessInfo
+                accessKind="noAccessButHasTraits"
+                traitKind={traitKind}
+                traitDesc={traitDescription}
+              />
+            )}
+            {isSsoUser && (
+              <AccessInfo
+                accessKind="ssoUserButHasTraits"
+                traitKind={traitKind}
+                traitDesc={traitDescription}
+              />
+            )}
+          </>
+        );
+      }
+
+      break;
+  }
+
+  return (
+    <Box maxWidth="700px">
+      <Header>Set Up Access</Header>
+      <HeaderSubtitle>{headerSubtitle}</HeaderSubtitle>
+      {infoContent}
+      <>
```

Comment:
No need for fragment. Maybe we should enable https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-no-useless-fragment.md?

Reply:
it actually didn't help me catch it (b/c the rule is `more than one children` 😞) but I still added the rule anyways